tutorial/write-custom-widget.md

```diff
@@ -107,3 +107,178 @@ Take care not to keep any important state in your renderer - animation tickers
 are well suited to that location but user state would not be. A widget that is
 hidden may have it's renderer destroyed and if it is shown again the new renderer
 must be able to reflect the same widget state.
+
+### Example Widget
+
+The following covers in detail the creation of a simple Widget and it's Renderer.
+
+#### Create a Widget
+
+So the widget requires the`widget.BaseWidget`. For example:
+
+```go
+type MyWidget struct {
+	widget.BaseWidget           // Inherit from BaseWidget
+	minSize           fyne.Size // The minimum size of the widget
+	text              string	// The text to display in the widget
+}
+```
+
+To construct the widget create a New<widgetname> function. The example we are using is called MyWidget so by convention the following constructor function name would be used:
+
+```go
+func NewMyWidget(W, H float32, text string) *MyWidget {
```

Comment:
Width and Height are controlled by the parent container and should not be passed in to a constructor.
The text parameter is enough for this example anyhow.